Lib/asyncio/streams.py

```diff
@@ -280,10 +283,16 @@ def connection_lost(self, exc):
         self._stream_writer = None
         self._transport = None
 
-    def data_received(self, data):
+    def get_buffer(self, sizehint):
+        if self._buffer is None:
+            size = self._default_buffer_size if sizehint == -1 else sizehint
```

Comment:
is sizehint is dynamic, this will keep it at a fixed size (which at the beginning might be small?)

Reply:
You are right, I should probably ignore that argument.
Fixed.